File: 1015-smallest-integer-divisible-by-k/1015-smallest-integer-divisible-by-k.py
class Solution(object):
    def smallestRepunitDivByK(self, k):
        if k%10 not in {1, 3, 7, 9}: return -1
        mod, ms= 0, set()
        for length in range(1, k+1):
            mod = (10*mod + 1)%k
            if mod == 0: return length
            if mod in ms: return -1
            ms.add(mod)
        return -1
        

File: 1015-smallest-integer-divisible-by-k/1015-smallest-integer-divisible-by-k.cpp
class Solution {
public:
    int smallestRepunitDivByK(int k) {
        if(k%10 != 1 && k%10 != 3 && k%10 != 7 && k%10 != 9) return -1;
        int m = 0;
        unordered_set<int> ms;
        for(int i = 1; i <= k; ++i) {
            m = (m*10+1)%k;
            if(m == 0) return i;
            if(ms.count(m)) return -1;
            ms.insert(m);
        }
        return -1;
    }
};